sdk/src/main/scala/io/horizen/certificatesubmitter/strategies/WithKeyRotationCircuitStrategy.scala

```diff
@@ -75,6 +77,13 @@ class WithKeyRotationCircuitStrategy[
 
     val schnorrKeysSignatures = getSchnorrKeysSignatures(state, status.referencedEpoch)
 
+    val sc2ScDataForCertificate: Option[Sc2ScDataForCertificate] =
+      sc2scConfig.canSendMessages match {
+        case true => Some(getDataForCertificateCreation(status.referencedEpoch, state, history, params))
```

Comment:
this method is and must be used only in this class (2 times: in message to sign and proof creation).
`Sc2ScDataForCertificate` result object contains `previousTopQualityCertificateHash` that is also used by this type of circuit. See `previousCertificateOption` object couple of lines above.

No sense to duplication this logic.